app/routes/talks.tsx

```diff
@@ -87,15 +97,182 @@ export const loader: LoaderFunction = async ({request}) => {
     checkValue: (value: unknown) => Array.isArray(value),
   })
 
-  const data: LoaderData = {talks}
+  const tags = Array.from(new Set(talks.flatMap(x => x.tags))).sort()
+
+  const data: LoaderData = {talks, tags}
   return json(data)
 }
 
+function Card({
+  tags,
+  title,
+  deliveries,
+  descriptionHTML,
+  resourceHTMLs,
+}: LoaderData['talks'][0]) {
+  const latestDate = deliveries
+    .filter(x => x.date)
+    .map(x => new Date(x.date as string))
+    .sort((l, r) => r.getTime() - l.getTime())[0] as Date
+
+  const isInFuture = latestDate.getTime() > Date.now()
+
+  // `software development` makes a terrible tag to render in this position
+  const tag = tags.filter(x => x.length <= 10)[0]
```

Comment:
Maybe we could sort all the tags (for all the talks) by the most frequently used and then choose the tag of this talk that is the most frequently used by the talks altogether and use that one.

Reply:
Yeah, that makes sense. Should we also use that to limit the number of options in the filter? Only show tags that are used `n` times? (need to find a nice number)

Reply:
Yup. We could just limit it to a total number of tags. Maybe the top 15-20?

Reply:
I'm now showing every tag that's being used by at least 2 talks. That results in 9 filterable tags.